torch/distributed/_device_mesh.py

```diff
@@ -381,8 +381,35 @@ def shape(self) -> Tuple[int, ...]:
         return tuple(self.mesh.shape)
 
     def get_rank(self) -> int:
+        """
+        Returns the current global rank.
+        """
         return get_rank()
 
+    def get_local_rank(
+        self, mesh_dim: Optional[Union[int, str]] = None
+    ) -> int:
+        """
+        Returns the local rank of the given mesh_dim.
+
+        Optional Args:
+            mesh_dim (str/int): it can be the name of the mesh dimension or the index
+            of the mesh dimension. Default is None.
+
+        Returns:
+            An integer denotes the local rank.
```

Comment:
shall we add a small example in the doc about a 2-D mesh, to describe the meaning of local_rank on mesh dimension?